src/compiler/checker.ts

```diff
@@ -13698,13 +13722,33 @@ namespace ts {
 
         function checkExternalModuleExports(node: SourceFile | ModuleDeclaration) {
             let moduleSymbol = getSymbolOfNode(node);
-            let links = getSymbolLinks(moduleSymbol);
+            let links: SymbolLinks = getSymbolLinks(moduleSymbol);
             if (!links.exportsChecked) {
                 let exportEqualsSymbol = moduleSymbol.exports["export="];
                 if (exportEqualsSymbol && hasExportedMembers(moduleSymbol)) {
                     let declaration = getDeclarationOfAliasSymbol(exportEqualsSymbol) || exportEqualsSymbol.valueDeclaration;
                     error(declaration, Diagnostics.An_export_assignment_cannot_be_used_in_a_module_with_other_exported_elements);
                 }
+                let exports = getExportsOfModule(moduleSymbol); // Checks for export * conflicts
+                for (let id in exports) {
+                    if (id === "__export") continue;
+                    if (!(exports[id].flags & SymbolFlags.Namespace || exports[id].flags & SymbolFlags.Interface) && exports[id].declarations.length > 1) { // 15.2.1.1 It is a Syntax Error if the ExportedNames of ModuleItemList contains any duplicate entries. (TS Exceptions: namespaces, function overloads, and interfaces)

```

Comment:
Good point about enums - they do merge, don't they. I'll add them to the conditional and to the test.